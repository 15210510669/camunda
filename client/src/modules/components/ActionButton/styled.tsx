/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import styled, {css} from 'styled-components';
import {IconButton} from 'modules/components/IconButton';

const Button = styled(IconButton)`
  ${({theme}) => {
    const colors = theme.colors.variables.editButton;

    return css`
      margin-left: 8px;
      z-index: 0;

      svg {
        margin-top: 4px;
      }

      &:disabled,
      &:disabled :hover {
        svg {
          color: ${colors.disabled.color};
          opacity: 0.5;
        }

        &:before {
          background-color: transparent;
        }
      }
    `;
  }}
`;

export {Button};
