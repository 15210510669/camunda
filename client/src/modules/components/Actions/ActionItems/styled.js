import styled, {css} from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';

import {ReactComponent as Retry} from 'modules/components/Icon/retry.svg';
import {ReactComponent as Stop} from 'modules/components/Icon/stop.svg';

export const Ul = themed(styled.ul`
  display: inline-flex;
  flex-direction: row;
  padding: 2px;
  margin: 0px;

  list-style-type: none;
  border-radius: 12px;

  background: ${({children}) =>
    // checks if children are components
    children.reduce((sum, next) => sum || !!next, false) &&
    themeStyle({
      dark: Colors.uiDark01,
      light: 'rgba(98, 98, 110, 0.15)'
    })};

  /* Border between single action items */
  & > li {
    &:not(:last-child) {
      border-right: solid 1px
        ${themeStyle({
          dark: 'rgba(250, 250, 250, 0.15)',
          light: 'rgb(98, 98, 110, 0.15)'
        })};
    }
  }

  /* Button border radius for correct focus style shapes*/

  /* only one child exists */
  & > li:first-child:last-child > button {
    border-radius: 12px;
  }

  & > li:first-child > button {
    border-radius: 12px 0 0 12px;
  }
  & > li:last-child > button {
    border-radius: 0 12px 12px 0;
  }
  & > li:not(:first-child):not(:last-child) > button {
    border-radius: 0px;
  }
`);

export const Button = themed(styled.button`
  display: flex;
  align-items: center;
  padding: 3px;

  background: none;
  border: none;
  border-radius: 12px;

  &:focus {
    outline: none;
    box-shadow: ${themeStyle({
      dark: `0 0 0 1px ${Colors.lightFocusInner},0 0 0 4px ${
        Colors.focusOuter
      }`,

      light: `0 0 0 1px ${Colors.darkFocusInner}, 0 0 0 4px ${
        Colors.focusOuter
      }`
    })};
  }
`);

export const iconStyle = css`
  opacity: 0.7;
  color: ${themeStyle({
    dark: '#ffffff',
    light: Colors.uiDark02
  })};
  cursor: pointer;
`;

// import and style any Icon accordingly
export const RetryIcon = themed(styled(Retry)`
  ${iconStyle};
`);

export const CancelIcon = themed(styled(Stop)`
  ${iconStyle};
`);
