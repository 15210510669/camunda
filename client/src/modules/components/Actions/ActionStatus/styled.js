import styled from 'styled-components';

import Spinner from 'modules/components/Spinner';

import {Colors, themed, themeStyle} from 'modules/theme';

export const ActionSpinner = themed(styled(Spinner)`
  margin: 0 5px;
 border: 3px solid ${({selected}) =>
   themeStyle({
     dark: '#ffffff',
     light: selected ? Colors.selections : Colors.badge02
   })};
    border-right-color: transparent;
  }
`);
