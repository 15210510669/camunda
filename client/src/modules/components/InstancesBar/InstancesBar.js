/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled.js';

function InstancesBar(props) {
  const {label, activeCount, incidentsCount, size} = props;
  const incidentsBarRatio =
    (100 * incidentsCount) / (activeCount + incidentsCount);

  const hasIncidents = incidentsCount > 0;
  const hasActive = activeCount > 0;

  return (
    <div className={props.className}>
      <Styled.Wrapper size={size}>
        <Styled.IncidentsCount hasIncidents={hasIncidents}>
          {incidentsCount}
        </Styled.IncidentsCount>
        <Styled.Label>{label}</Styled.Label>
        <Styled.ActiveCount hasActive={hasActive}>
          {activeCount}
        </Styled.ActiveCount>
      </Styled.Wrapper>
      <Styled.BarContainer size={size}>
        <Styled.Bar hasActive={hasActive} />
        <Styled.IncidentsBar
          style={{
            width: `${incidentsBarRatio}%`
          }}
        />
      </Styled.BarContainer>
    </div>
  );
}

InstancesBar.propTypes = {
  label: PropTypes.string,
  activeCount: PropTypes.number.isRequired,
  incidentsCount: PropTypes.number.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']).isRequired
};

export default InstancesBar;
