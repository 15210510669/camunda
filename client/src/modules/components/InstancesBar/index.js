/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled.js';

function InstancesBar(props) {
  const {label, activeCount, incidentsCount, size, barHeight} = props;
  const incidentsBarRatio =
    (100 * incidentsCount) / (activeCount + incidentsCount);

  const hasIncidents = incidentsCount > 0;
  const hasActive = activeCount > 0;

  return (
    <div className={props.className}>
      <Styled.Wrapper size={size}>
        <Styled.IncidentsCount
          data-test="incident-instances-badge"
          hasIncidents={hasIncidents}
        >
          {incidentsCount}
        </Styled.IncidentsCount>
        <Styled.Label>{label}</Styled.Label>
        {activeCount >= 0 && (
          <Styled.ActiveCount
            data-test="active-instances-badge"
            hasActive={hasActive}
          >
            {activeCount}
          </Styled.ActiveCount>
        )}
      </Styled.Wrapper>
      <Styled.BarContainer height={barHeight}>
        <Styled.Bar hasActive={hasActive} />
        <Styled.IncidentsBar
          style={{
            width: `${incidentsBarRatio}%`,
          }}
        />
      </Styled.BarContainer>
    </div>
  );
}

InstancesBar.propTypes = {
  label: PropTypes.string,
  activeCount: PropTypes.number,
  incidentsCount: PropTypes.number.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']).isRequired,
  barHeight: PropTypes.number.isRequired,
};

export {Wrapper, Bar} from './styled';
export default InstancesBar;
