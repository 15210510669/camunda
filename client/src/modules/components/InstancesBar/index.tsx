/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import * as Styled from './styled';

type Props = {
  label?: string;
  activeCount?: number;
  incidentsCount: number;
  className?: string;
  size: 'small' | 'medium' | 'large';
  barHeight: number;
};

function InstancesBar(props: Props) {
  const {label, activeCount, incidentsCount, size, barHeight} = props;
  const incidentsBarRatio =
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    (100 * incidentsCount) / (activeCount + incidentsCount);

  const hasIncidents = incidentsCount > 0;
  // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
  const hasActive = activeCount > 0;

  return (
    <div className={props.className}>
      <Styled.Wrapper size={size}>
        <Styled.IncidentsCount
          data-testid="incident-instances-badge"
          hasIncidents={hasIncidents}
        >
          {incidentsCount}
        </Styled.IncidentsCount>
        <Styled.Label data-testid="incident-message">{label}</Styled.Label>
        {/* @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'. */}
        {activeCount >= 0 && (
          <Styled.ActiveCount
            data-testid="active-instances-badge"
            hasActive={hasActive}
          >
            {activeCount}
          </Styled.ActiveCount>
        )}
      </Styled.Wrapper>
      <Styled.BarContainer height={barHeight}>
        <Styled.Bar hasActive={hasActive} />
        <Styled.IncidentsBar
          data-testid="incidents-bar"
          style={{
            width: `${incidentsBarRatio}%`,
          }}
        />
      </Styled.BarContainer>
    </div>
  );
}

export {Wrapper, Bar} from './styled';
export default InstancesBar;
