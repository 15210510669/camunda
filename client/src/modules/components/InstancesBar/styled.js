/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import styled, {css} from 'styled-components';

const FONT_STYLES = {
  small: css`
    font-weight: 400;
    font-size: 13px;
  `,
  medium: css`
    font-weight: 600;
    font-size: 14px;
  `,
  large: css`
    font-weight: 600;
    font-size: 30px;
  `,
};

const Wrapper = styled.div`
  ${({theme, size}) => {
    const colors = theme.colors.modules.instancesBar.wrapper;

    return css`
      display: flex;
      padding: 0;
      color: ${colors.color};
      font-family: IBMPlexSans;
      line-height: 1.71;
      ${FONT_STYLES[size]}
    `;
  }}
`;

const greyTextStyle = ({theme}) => {
  const colors = theme.colors.modules.instancesBar.greyTextStyle;

  return css`
    color: ${colors.color};
    opacity: 0.55;
  `;
};

const IncidentsCount = styled.div`
  ${({theme, hasIncidents}) => {
    return css`
      min-width: 96px;
      ${hasIncidents
        ? css`
            color: ${theme.colors.incidentsAndErrors};
          `
        : greyTextStyle}
    `;
  }}
`;

const ActiveCount = styled.div`
  ${({theme, hasActive}) => {
    return css`
      margin-left: auto;
      width: 139px;
      text-align: right;
      ${hasActive
        ? css`
            color: ${theme.colors.allIsWell};
            opacity: 0.8;
          `
        : greyTextStyle}
    `;
  }}
`;

const Label = styled.div`
  ${({theme}) => {
    const opacity = theme.opacity.modules.instancesBar.label;

    return css`
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      opacity: ${opacity};
    `;
  }}
`;

const BarContainer = styled.div`
  ${({height}) => {
    return css`
      position: relative;
      > div {
        height: ${height}px;
      }
    `;
  }}
`;

const Bar = styled.div`
  ${({theme, hasActive}) => {
    const opacity = theme.opacity.modules.instancesBar.bar.active;

    return css`
      ${hasActive
        ? css`
            background: ${theme.colors.allIsWell};
            opacity: 0.8;
          `
        : css`
            background: ${theme.colors.badge02};
            opacity: ${opacity};
          `}
    `;
  }}
`;

const IncidentsBar = styled.div`
  ${({theme}) => {
    return css`
      position: absolute;
      top: 0;
      background: ${theme.colors.incidentsAndErrors};
    `;
  }}
`;

export {
  Wrapper,
  IncidentsCount,
  ActiveCount,
  Label,
  BarContainer,
  Bar,
  IncidentsBar,
};
