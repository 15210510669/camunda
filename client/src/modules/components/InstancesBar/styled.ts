/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import styled, {css, ThemedInterpolationFunction} from 'styled-components';

const FONT_STYLES = {
  small: css`
    font-weight: 400;
    font-size: 13px;
  `,
  medium: css`
    font-weight: 600;
    font-size: 14px;
  `,
  large: css`
    font-weight: 600;
    font-size: 30px;
  `,
} as const;

type WrapperProps = {
  size: 'small' | 'medium' | 'large';
};

const Wrapper = styled.div<WrapperProps>`
  ${({theme, size}) => {
    return css`
      display: flex;
      padding: 0;
      color: ${theme.colors.text02};
      font-family: IBM Plex Sans;
      line-height: 1.71;
      ${FONT_STYLES[size]}
    `;
  }}
`;

const greyTextStyle: ThemedInterpolationFunction = ({theme}) => {
  const colors = theme.colors.modules.instancesBar.greyTextStyle;

  return css`
    color: ${colors.color};
    opacity: 0.55;
  `;
};

type IncidentsCountProps = {
  hasIncidents?: boolean;
};

const IncidentsCount = styled.div<IncidentsCountProps>`
  ${({theme, hasIncidents}) => {
    return css`
      min-width: 96px;
      ${hasIncidents
        ? css`
            color: ${theme.colors.incidentsAndErrors};
          `
        : greyTextStyle}
    `;
  }}
`;

type ActiveCountProps = {
  hasActive?: boolean;
};

const ActiveCount = styled.div<ActiveCountProps>`
  ${({theme, hasActive}) => {
    return css`
      margin-left: auto;
      width: 139px;
      text-align: right;
      ${hasActive
        ? css`
            color: ${theme.colors.allIsWell};
            opacity: 0.8;
          `
        : greyTextStyle}
    `;
  }}
`;

const Label = styled.div`
  ${({theme}) => {
    const opacity = theme.opacity.modules.instancesBar.label;

    return css`
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      opacity: ${opacity};
    `;
  }}
`;

type BarContainerProps = {
  height: number;
};

const BarContainer = styled.div<BarContainerProps>`
  ${({height}) => {
    return css`
      position: relative;
      > div {
        height: ${height}px;
      }
    `;
  }}
`;

type BarProps = {
  hasActive?: boolean;
};

const Bar = styled.div<BarProps>`
  ${({theme, hasActive}) => {
    const opacity = theme.opacity.modules.instancesBar.bar.active;

    return css`
      ${hasActive
        ? css`
            background: ${theme.colors.allIsWell};
            opacity: 0.8;
          `
        : css`
            background: ${theme.colors.badge02};
            opacity: ${opacity};
          `}
    `;
  }}
`;

const IncidentsBar = styled.div`
  ${({theme}) => {
    return css`
      position: absolute;
      top: 0;
      background: ${theme.colors.incidentsAndErrors};
    `;
  }}
`;

export {
  Wrapper,
  IncidentsCount,
  ActiveCount,
  Label,
  BarContainer,
  Bar,
  IncidentsBar,
};
