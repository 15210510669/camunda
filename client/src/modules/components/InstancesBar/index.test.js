/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';
import InstancesBar from './index';
import * as Styled from './styled';

describe('InstancesBar', () => {
  it('should display the right data', () => {
    const node = shallow(
      <InstancesBar
        incidentsCount={10}
        label="someLabel"
        activeCount={8}
        barHeight={5}
        size="small"
      />
    );

    expect(node.find(Styled.IncidentsCount).text()).toBe('10');
    expect(node.find(Styled.Label).text()).toBe('someLabel');
    expect(node.find(Styled.ActiveCount).text()).toBe('8');
    expect(node.find(Styled.Bar)).toExist();
    expect(node.find(Styled.IncidentsBar).props().style.width).toContain(
      (10 * 100) / 18
    );
    expect(node).toMatchSnapshot();
  });
});
