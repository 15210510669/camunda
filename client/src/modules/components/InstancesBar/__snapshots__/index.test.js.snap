// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesBar should display the right data 1`] = `
<div>
  <Themed(styled.div)
    size="small"
  >
    <Themed(styled.div)
      data-test="incident-instances-badge"
      hasIncidents={true}
    >
      10
    </Themed(styled.div)>
    <Themed(styled.div)>
      someLabel
    </Themed(styled.div)>
    <Themed(styled.div)
      data-test="active-instances-badge"
      hasActive={true}
    >
      8
    </Themed(styled.div)>
  </Themed(styled.div)>
  <styled.div
    height={5}
  >
    <Themed(styled.div)
      hasActive={true}
    />
    <styled.div
      style={
        Object {
          "width": "55.55555555555556%",
        }
      }
    />
  </styled.div>
</div>
`;
